### PR TITLE
Enter accessibility

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
The only way to get into keyboard accessibility mode used to be by shift clicking on the workspace.

### Proposed Changes
1. Add a checkbox on the playground for inserting into keyboard navigation mode.
2. Add an action for toggling keyboard navigation. Currently mapped to hitting control k
3. Clicking on the workspace no longer takes us out of keyboard navigation. Instead keyboard navigation is turned off by hitting control k once keyboard navigation is on.

### Reason for Changes
Shift clicking wasn't a very accessible way to turn on keyboard navigation.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/gesture.js [hunk 1]
```diff
@@ -498,9 +498,12 @@ Blockly.Gesture.prototype.doStart = function(e) {
   Blockly.Tooltip.block();
 
   if (this.targetBlock_) {
-    this.targetBlock_.select();
     if (!this.targetBlock_.isInFlyout && e.shiftKey) {
       Blockly.navigation.enableKeyboardAccessibility();
+      var ws = this.creatorWorkspace_;
+      ws.cursor.setLocation(Blockly.navigation.getTopNode(this.targetBlock_));
```
Comment (samelhusseini) [inline core/gesture.js]: do we need this var?

File: core/keyboard_nav/key_map.js [hunk 2]
```diff
@@ -124,13 +124,35 @@ Blockly.user.keyMap.serializeKeyEvent = function(e) {
   return key;
 };
 
+/**
+ * Create the serialized key code that will be used in the key map.
+ * @param {!number} keyCode Number code representing the key.
+ * @param {!Array<string>} modifiers List of modifiers to be used with the key.
+ *     All valid modifiers can be found in the Blockly.user.keyMap.modifiers list.
```
Comment (samelhusseini) [inline core/keyboard_nav/key_map.js]: Blockly.user.keyMap.modifierKeys

File: core/keyboard_nav/key_map.js [hunk 3]
```diff
@@ -124,13 +124,35 @@ Blockly.user.keyMap.serializeKeyEvent = function(e) {
...
+};
+
 /**
  * Creates the default key map.
  * @return {!Object<string,Blockly.Action>} An object holding the default key
  *     to action mapping.
  */
 Blockly.user.keyMap.createDefaultKeyMap = function() {
   var map = {};
+  var controlK = Blockly.user.keyMap.createSerializeKey(Blockly.utils.KeyCodes.K, ['Control']);
```
Comment (samelhusseini) [inline core/keyboard_nav/key_map.js]: Can we create constants for these modifiers, strings are error prone.

File: core/keyboard_nav/navigation.js [hunk 4]
```diff
@@ -1116,6 +1123,12 @@ Blockly.navigation.ACTION_TOOLBOX = new Blockly.Action(
 Blockly.navigation.ACTION_EXIT = new Blockly.Action(
     Blockly.navigation.actionNames.EXIT, 'Close the current modal, such as a toolbox or field editor.');
 
+/**
+ * The action to toggle keyboard navigation mode on and off.
+ */
```
Comment (samelhusseini) [inline core/keyboard_nav/navigation.js]: missing type annotation.

Comment (samelhusseini) [reply core/keyboard_nav/navigation.js]: `` * @type {!Blockly.Action}``

File: core/keyboard_nav/navigation.js [hunk 5]
```diff
@@ -882,7 +886,10 @@ Blockly.navigation.onKeyPress = function(e) {
  * @package
  */
 Blockly.navigation.onBlocklyAction = function(action) {
-  if (Blockly.navigation.currentState_ === Blockly.navigation.STATE_WS) {
+  if (action.name == Blockly.navigation.actionNames.TOGGLE_KEYBOARD_NAV) {
```
Comment (samelhusseini) [inline core/keyboard_nav/navigation.js]: For the sake of consistency with others, use ``===``

File: core/keyboard_nav/navigation.js [hunk 6]
```diff
@@ -860,15 +857,22 @@ Blockly.navigation.onKeyPress = function(e) {
...
+          actionHandled = Blockly.navigation.onBlocklyAction(action);
+        }
+      // If in readonly mode only handle valid actions.
+      } else if (Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
         actionHandled = Blockly.navigation.onBlocklyAction(action);
       }
-    } else if (Blockly.navigation.READONLY_ACTION_LIST.indexOf(action) > -1) {
-      actionHandled = Blockly.navigation.onBlocklyAction(action);
+    // If not in accessibility mode only hanlde turning on keyboard navigation.
+    } else if (action.name == Blockly.navigation.actionNames.TOGGLE_KEYBOARD_NAV) {
```
Comment (samelhusseini) [inline core/keyboard_nav/navigation.js]: use ``===``, for consistency.

**Comment by alschmiedt** [comment]
Ok think I got all of them

File: core/keyboard_nav/key_map.js [hunk 7]
```diff
@@ -124,13 +129,37 @@ Blockly.user.keyMap.serializeKeyEvent = function(e) {
   return key;
 };
 
+/**
+ * Create the serialized key code that will be used in the key map.
+ * @param {!number} keyCode Number code representing the key.
+ * @param {!Array<string>} modifiers List of modifiers to be used with the key.
+ *     All valid modifiers can be found in the Blockly.user.keyMap.modifierKeys.
+ * @return {string} The serialized key code for the given modifiers and key.
+ */
```
Comment (samelhusseini) [inline core/keyboard_nav/key_map.js]: public on purpose?

File: core/keyboard_nav/key_map.js [hunk 8]
```diff
@@ -124,13 +129,37 @@ Blockly.user.keyMap.serializeKeyEvent = function(e) {
...
+Blockly.user.keyMap.createSerializeKey = function(keyCode, modifiers) {
```
Comment (samelhusseini) [inline core/keyboard_nav/key_map.js]: nit: can this be ``createSerializedKey`` instead?

File: core/keyboard_nav/key_map.js (hunk 7 above)
Comment (alschmiedt) [reply core/keyboard_nav/key_map.js]: Yeah, it is so developers can easily create a keycode for something like control K and then set it for an action.

Comment (samelhusseini) [reply core/keyboard_nav/key_map.js]: Sounds good.

**Pull request merged**